lib/dry/types/hash/schema.rb

```diff
@@ -4,15 +4,15 @@ class Hash < Definition
       class Schema < Hash
         attr_reader :member_types
 
-        def initialize(primitive, options = {})
+        def initialize(_primitive, options)
           @member_types = options.fetch(:member_types)
           super
         end
 
         def call(hash, meth = :call)
           member_types.each_with_object({}) do |(key, type), result|
             if hash.key?(key)
-              result[key] = type.__send__(meth, hash[key])
+              result[key] = type.public_send(meth, hash.fetch(key))
```

Comment:
> Can we leave it for now?

Fair enough

> The semantics of public_send are more correct in this use case and the performance difference is just barely statistically significant

In an isolated benchmark, yes, but all those tiny little details add up easily. Let's stick to `public_send` for now and I'll look into total performance impact at the end, once we're done with refactorings, it may, or may not be an issue. We'll see :)